compiler/rustc_target/src/spec/thumbv4t_none_eabi.rs

```diff
@@ -39,13 +47,20 @@ pub fn target() -> Target {
             // minimum extra features, these cannot be disabled via -C
             features: "+soft-float,+strict-align".into(),
 
+            panic_strategy: PanicStrategy::Abort,
+            relocation_model: RelocModel::Static,
+            // suggested from thumb_base, rust-lang/rust#44993.
+            emit_debug_gdb_scripts: false,
+            // suggested from thumb_base, with no-os gcc/clang use 8-bit enums
+            c_enum_min_bits: 8,
+
             main_needs_argc_argv: false,
 
             // don't have atomic compare-and-swap
             atomic_cas: false,
             has_thumb_interworking: true,
 
-            ..super::thumb_base::opts()
+            ..Default::default()
```

Comment:
I'm a little surprised from the PR description that this doesn't just manually set the frame pointer to "may omit" which `thumb_base` gets wrong for this specific target, but instead decides not to use `thumb_base` at all.

Reply:
Somehow that didn't even occur to me compared to "give me the real defaults".

Reply:
I think I'd prefer to see it still use `thumb_base` and override the changed field, rather than this approach. Unless there's a reason why this specific target is likely to diverge from `thumb_base` more often than not. I'm not super familiar with what we normally do for target specifications though, so others may disagree.

Reply:
I can set the field here, but either way I'd prefer to not use thumb_base. Then as the target maintainer i have to try and pay attention to changes to an additional thing. Also as a user a person would have to look in 3 places instead of just 2 to understand all the details.

Reply:
That's understandable, but at the same time, if there is a change to `thumb_base` that would be applicable here then it could easily be missed. Ultimately, this is a trade-off that's inherent to how we define these target specs, and I'm not sure it makes sense to be inconsistent about inheriting from base specs on the basis of the preference of the target maintainer, but I don't feel strongly about this. I'd like to see what others think.